tools/setup/app.nim

```diff
@@ -0,0 +1,138 @@
+import std/os
+import std/osproc
+import std/httpclient
+import std/strutils
+import pkg/chronicles
+import pkg/questionable
+import pkg/ethers
+import ./print
+import ./networkconfig
+
+type App* = ref object
+  configLines: seq[string]
+  networkConfig: ?ArchivistNetwork
+
+  ethAddress: string
+  # if we have networkConfig AND storage mode (not manual) is selected
+  # then run CIRDL
+  storageModeSelected*: bool
+  # if testnet or devnet, display faucet links
+  faucetLinkNetwork*: string
+  # if set, display webui link
+  webUi*: bool
+
+proc writeConfigLine*(app: App, line: string) =
+  app.configLines.add(line)
+
+proc fetchNetworkConfig*(app: App, network: string): ArchivistNetwork =
+  if isNone(app.networkConfig):
+    info "Fetching network information...", network
+    app.networkConfig = some getNetworkConfig(network)
+  return !app.networkConfig
```

Comment:
Just a suggestion:
```suggestion
  without var networkConfig =? app.networkConfig:
    info "Fetching network information...", network
    networkConfig = getNetworkConfig(network)
    app.networkConfig = some networkConfig
  return networkConfig
```
I try to avoid the use of `!` in production code.